lib/signs_ui/signs/expiration.ex

```diff
@@ -0,0 +1,84 @@
+defmodule SignsUI.Signs.Expiration do
+  @moduledoc """
+  Handle cleaning up settings in the configuration that have since expired
+  """
+  use GenServer
+
+  @type state :: %{
+          time_fetcher: (() -> DateTime.t()),
+          loop_ms: integer(),
+          sign_state_server: GenServer.server()
+        }
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(opts \\ []) do
+    name = Keyword.get(opts, :name, __MODULE__)
+
+    GenServer.start_link(__MODULE__, opts, name: name)
+  end
+
+  @spec init(Keyword.t()) :: {:ok, state()}
+  def init(opts) do
+    time_fetcher = Keyword.get(opts, :time_fetcher, fn -> DateTime.utc_now() end)
+    loop_ms = Keyword.get(opts, :loop_ms, 5_000)
+    sign_state_server = Keyword.get(opts, :sign_state_server, SignsUI.Signs.State)
+    schedule_loop(self(), loop_ms)
+    {:ok, %{time_fetcher: time_fetcher, loop_ms: loop_ms, sign_state_server: sign_state_server}}
+  end
+
+  @spec handle_info(:process_expired, state()) :: state()
+  def handle_info(:process_expired, state) do
+    expire_signs(state.time_fetcher, state.sign_state_server)
+
+    schedule_loop(self(), state.loop_ms)
+
+    {:noreply, state}
+  end
+
+  @spec expire_signs((() -> DateTime.t()), GenServer.server()) :: :ok
+  def expire_signs(time_fetcher, sign_state_server) do
+    current_dt = time_fetcher.()
+
+    expired_signs =
+      sign_state_server
+      |> SignsUI.Signs.State.get_all()
+      |> Enum.flat_map(&expire_single_sign(&1, current_dt))
+      |> Enum.into(%{})
+
+    if expired_signs != %{} do
+      SignsUI.Signs.State.update_some(sign_state_server, expired_signs)
+    else
+      :ok
+    end
+  end
+
+  @spec expire_single_sign(
+          {SignsUI.Signs.Sign.id(), SignsUI.Signs.Sign.t()},
```

Comment:
If you `alias SignsUI.Signs.Sign` in this file, and then refer to `Sign.id()`, `Sign.t()`, and `%Sign{..}` does it neaten this up a little bit? It's kind of hard for me to follow what's going on with all the line breaks and indents.

Reply:
Oh, I had thought that to myself when I was working on it but never got around to actually doing it. I probably should.